### PR TITLE
Add arrowGlacier HF

### PR DESCRIPTION
**Pull request by jochem-brouwer**
This PR adds the arrow glacier HF, which delays the difficulty bomb.

Since we have no test cases, do not merge yet.

### CODE REVIEW

**Comment by holgerd77** [comment]
I guess at some point we should also get over the finish line with this. Any updates here? What's the current status? 🤔

**Comment by jochem-brouwer** [comment]
If we want we can merge this and then at a later point run it against ethereum/tests? I can try running the test locally to see if they pass, if thats the case then we can also merge.

**Comment by holgerd77** [comment]
Ok, maybe let's then wait for the official `ethereum/tests` release, guess that can't be too far away, and some in between step we should also update to [v10.1](https://github.com/ethereum/tests/releases/tag/v10.1) since this release comes with some significant structural changes (legacy tests moved to own repo).

@jochem-brouwer Can you do rebase in between and bring this up-to-date so that we can keep up here a bit with keeping this close to `master`?

File: packages/common/src/chains/mainnet.json [hunk 1]
```diff
@@ -80,15 +80,20 @@
       "block": 12965000, 
       "forkHash": "0xb715077d"
     },
+    {
+      "name": "arrowGlacier",
+      "block": 13773000,
+      "forkHash": null
```
Comment (holgerd77) [inline packages/common/src/chains/mainnet.json:86]: Fork hash still missing here, can be taken from https://github.com/ethereum/execution-specs/blob/master/network-upgrades/mainnet-upgrades/arrow-glacier.md

**Comment by holgerd77** [comment]
Will `mainnet` be the only fork here or will there be a test fork on `ropsten`? Or would this not working out due to differing practical TD values anyhow? 🤔

**Comment by jochem-brouwer** [comment]
Ok this is weird, the tests do not pass, I tried locally. To test;

``
cd packages/ethereum-tests
git checkout develop
git pull
``

This pulls the latest data from the repo, the difficulty tests are not added yet.

Now to `/block/test/difficulty.spec.ts` add this to `hardforkTestData` (line 37)

```
arrowGlacier: require('../../ethereum-tests/DifficultyTests/dfArrowGlacier/difficultyArrowGlacier.json').difficultyArrowGlacier.ArrowGlacier
```

Only running these tests;

```
1..10776
# tests 10776
# pass  9896
# fail  880
```

**Comment by holgerd77** [comment]
Yes, can confirm.

Also ran this locally, when digging a bit deeper one sees that every second test with `"parentUncles" : "0x00"` is failing.

Not sure yet what is happening, the format of this uncles field changed substantially (there was a full hash before), will ask on `ethereum/tests`.

**Comment by holgerd77** [comment]
The semantics of the `uncles` field has changed, see [here](https://github.com/ethereum/tests/pull/970/files#diff-cba7a4cc7760cb0cd6dccb9cbfdb6423c5384b91295ed475c1cd21374076ffac).

If one is replacing `0x00` (which now stands for "no uncles") with an just an empty field (which translates for our instantiation to "instantiate with no uncles", this works again.

Will push.

**Comment by jochem-brouwer** [comment]
@holgerd77 I see, but I don't understand how this passes. I don't see anything in your commits which handles this `parentUncles` field, so I do not understand why this passes. Is it an edited tests file?

(Small sidenote: we should at some point make these block tests not use the copied JSON files from ethereum/tests, but directly read them from ethereum/tests)

**Comment by holgerd77** [comment]
Yes, edited test file.

**Comment by jochem-brouwer** [comment]
Oh right, I can't approve my own PR 😂

**Review by holgerd77 — APPROVED** [review]
LGTM and x-approved by @jochem-brouwer, will merge.

**Pull request merged**